File: app/src/main/java/com/sevtinge/hyperceiler/module/hook/screenshot/UnlockMinimumCropLimit.java
/*
 * This file is part of HyperCeiler.

 * HyperCeiler is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License.

 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.

 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.

 * Copyright (C) 2023-2024 HyperCeiler Contributions
 */
package com.sevtinge.hyperceiler.module.hook.screenshot;

import com.sevtinge.hyperceiler.module.base.BaseHook;
import com.sevtinge.hyperceiler.module.base.dexkit.DexKit;
import com.sevtinge.hyperceiler.module.base.dexkit.IDexKitList;

import org.luckypray.dexkit.DexKitBridge;
import org.luckypray.dexkit.query.FindMethod;
import org.luckypray.dexkit.query.matchers.MethodMatcher;
import org.luckypray.dexkit.result.MethodDataList;

import java.lang.reflect.AnnotatedElement;
import java.lang.reflect.Method;
import java.util.ArrayList;

public class UnlockMinimumCropLimit extends BaseHook {
    @Override
    public void init() throws NoSuchMethodException {
        ArrayList<Method> methods = DexKit.getDexKitBridge("MinimumCropLimit", new IDexKitList() {
            @Override
            public ArrayList<AnnotatedElement> dexkit(DexKitBridge bridge) throws ReflectiveOperationException {
                MethodDataList methodData = bridge.findMethod(FindMethod.create()
                        .matcher(MethodMatcher.create()
                                .returnType(int.class)
                                .paramCount(0)
                                .usingNumbers(0.5f, 200)
                        )
                );
                return DexKit.toElementList(methodData, lpparam.classLoader);
            }
        }).toMethodList();
        for (Method method : methods) {
            hookMethod(method, new MethodHook() {
                @Override
                protected void before(MethodHookParam param) throws Throwable {
                    param.setResult(0);
                }
            });
        }
    }
}

File: app/src/main/java/com/sevtinge/hyperceiler/module/hook/screenshot/UnlockMinimumCropLimit.kt
package com.sevtinge.hyperceiler.module.hook.screenshot

import android.annotation.SuppressLint
import com.github.kyuubiran.ezxhelper.EzXHelper
import com.github.kyuubiran.ezxhelper.HookFactory.`-Static`.createHook
import com.sevtinge.hyperceiler.module.base.BaseHook
import com.sevtinge.hyperceiler.utils.DexKit.dexKitBridge
import java.lang.reflect.Modifier

@SuppressLint("StaticFieldLeak")
object UnlockMinimumCropLimit : BaseHook() {
    private val mScreenCropViewMethod by lazy {
        dexKitBridge.findMethod {
            matcher {
                usingNumbers(0.5f, 200)
                returnType = "int"
                paramCount = 0
                modifiers = Modifier.PRIVATE
            }
        }.firstOrNull()?.getMethodInstance(EzXHelper.safeClassLoader)
    }


    override fun init() {
        mScreenCropViewMethod!!.createHook {
            returnConstant(0)
        }
    }
}
